server/client.go

```diff
@@ -5511,6 +5513,14 @@ func (c *client) closeConnection(reason ClosedState) {
 		}
 	}
 
+	// Now that we are done with subscriptions, clear the field so that the
+	// connection can be released and gc'ed.
+	if kind == CLIENT || kind == LEAF {
```

Comment:
FYI when we set this to nil in saveClosedClient it was a much broader set of connection types. Do we think this might cause any GC issues on non client/leaf connections?

Reply:
No because we invoked saveClosedClient() only for CLIENT and LEAF: https://github.com/nats-io/nats-server/blob/02bd28438a538c9182ee1ebf2b04d29dc4842483/server/client.go#L1828

Reply:
Sorry was just looking at the internal if..